### ISSUE TITLE
otp 25 upgrade ssl_opts issue

### ISSUE THREAD

**Issue by dvic** [elixir-ecto/postgrex#602]
Since we upgraded to OTP 25 we're getting 

```
failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification)
 connection requires a valid client certificate
```

Before this upgrade our `ssl_opts` config was working fine. Anyone else encountered this issue?

**Comment by greg-rychlewski** [comment]
@dvic Are you still having this issue? Would you be able to provide any details about your `ssl_opts` ?

**Comment by dvic** [comment]
I have to investigate this further, because some projects I upgraded did not suffer from this problem. I'll get back with more details.

**Comment by dvic** [comment]
I still have no idea why this was happening on this particular project but OTP 25.0.1. works now. Maybe it was fixed by the following PR: https://github.com/erlang/otp/pull/6037

**Comment by gesta** [comment]
I've been able to fix this error, occurring when attempting to connect to a Google Cloud SQL instance by restricting the [versions](https://www.erlang.org/doc/man/ssl.html#type-common_option) options in `ssl_opts` to `[:"tlsv1.2"]`.
While debugging I noticed an initial handshake attempt over TLS v1.3, then downgrading to a handshake over TLS v1.2 and then an error being returned. After restricting to TLS v1.2 only, the initial handshake attempt is successful and a connection is established afterwards.
NB: Google Cloud SQL supports only TLS versions 1.0, 1.1 and 1.2.
